core/dictgen/src/rootcling_impl.cxx

```diff
@@ -4233,9 +4253,46 @@ int RootClingMain(int argc,
 
    ROOT::TMetaUtils::SetPathsForRelocatability(clingArgs);
 
+   // FIXME: This line is from TModuleGenerator, but we can't reuse this code
+   // at this point because TModuleGenerator needs a CompilerInstance (and we
+   // currently create the arguments for creating said CompilerInstance).
+   bool isPCH = (dictpathname == "allDict.cxx");
+   StringRef moduleName;
+   // Adding -fmodules to the args will break lexing with __CINT__ defined,
+   // and we actually do lex with __CINT__ and reuse this variable later,
+   // we have to copy it now.
```

Comment:
IIRC, @Axel-Naumann had a solution how to fix this by instrumenting the PP callbacks. Is it worth mentioning it as a FIXME here?

Reply:
Hmm, code with __CINT__ doesn't parse into an AST but only lexes, so I think we can't do anything here?